submission_worker.py

```diff
@@ -0,0 +1,309 @@
+        with open(download_location, 'w') as f:
+            f.write(response.content)
+        # extract zip file
+        zip_ref = zipfile.ZipFile(download_location, 'r')
+        zip_ref.extractall(extract_location)
+        zip_ref.close()
+        # delete zip file
+        try:
+            os.remove(download_location)
+        except Exception as e:
+            print 'Failed to remove zip file {}, error {}'.format(download_location, e)
+
+
+def create_dir(directory):
+    '''
+        Creates a directory if it does not exists
+    '''
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def create_dir_as_python_package(directory):
+    '''
+        Create a directory and then makes it a python
+        package by creating `__init__.py` file.
+    '''
+    create_dir(directory)
+    init_file_path = join(directory, '__init__.py')
+    with open(init_file_path, 'w') as init_file:        # noqa
+        # create empty file
+        pass
+
+
+def extract_challenge_data(challenge, phases):
+    '''
+        * Expects a challenge object and an array of phase object
+        * Extracts `evaluation_script` for challenge and `annotation_file` for each phase
+
+    '''
+
+    challenge_data_directory = CHALLENGE_DATA_DIR.format(challenge_id=challenge.id)
+    evaluation_script_url = challenge.evaluation_script
+    # create challenge directory as package
+    create_dir_as_python_package(challenge_data_directory)
+
+    challenge_zip_file = join(challenge_data_directory, 'challenge_{}.zip'.format(challenge.id))
+    download_and_extract_file(evaluation_script_url, challenge_zip_file, challenge_data_directory)
+
+    phase_data_base_directory = PHASE_DATA_BASE_DIR.format(challenge_id=challenge.id)
+    create_dir(phase_data_base_directory)
+
+    for phase in phases:
+        phase_data_directory = PHASE_DATA_DIR.format(challenge_id=challenge.id, phase_id=phase.id)
+        # create phase directory
+        create_dir(phase_data_directory)
+        phase_zip_file = join(phase_data_directory, 'phase_{}.zip'.format(phase.id))
+        annotation_file_url = phase.test_annotation
+        download_and_extract_file(annotation_file_url, phase_zip_file, phase_data_directory)
+
+    # import the challenge after everything is finished
+    challenge_module = importlib.import_module(CHALLENGE_IMPORT_STRING.format(challenge_id=challenge.id))
+    EVALUATION_SCRIPTS[challenge.id] = challenge_module
+
+
+def load_active_challenges():
+    '''
+         * Fetches active challenges and corresponding active phases for it.
+    '''
+    q_params = {}
+    q_params['start_date__lt'] = timezone.now()
+    q_params['end_date__gt'] = timezone.now()
+
+    # make sure that the challenge base directory exists
+    create_dir_as_python_package(CHALLENGE_DATA_BASE_DIR)
+
+    active_challenges = Challenge.objects.filter(**q_params)
+
+    for challenge in active_challenges:
+        phases = challenge.challengephase_set.all()
```

Comment:
ChallengePhase does Foreign Key on Challenge, with no related_name, hence when fetching phase for a particular challenge I need to use `challengephase_set`.